ui/src/funding/contract.ts

```diff
@@ -1,25 +1,53 @@
-import type { BigNumber, ContractTransaction, Signer } from "ethers";
+import type { ethers, BigNumber, ContractTransaction, Signer } from "ethers";
+
+import Big from "big.js";
+
 import {
   Erc20Pool,
   Erc20Pool__factory as PoolFactory,
   ERC20,
   ERC20__factory as Erc20Factory,
 } from "radicle-contracts/build/contract-bindings/ethers";
 
-// TODO(nuno): make the contract addresses configurable/env-dependant
-
-export const POOL_ADDRESS: string =
-  "0x37c10d847bf9e708add451bf2f80c1297d7aa691";
+import * as ethereum from "../ethereum";
+
+const addresses = {
+  pool: {
+    local: "0x37c10d847bf9e708add451bf2f80c1297d7aa691",
+    ropsten: "0xEc5bfca987C5FAA6d394044793f0aD6C9A85Da76",
+  },
+  dai: {
+    local: "0xf34a89802590f944e3de71b1f74d66ed1bafc9cd",
+    ropsten: "0xD069f9Cbe64979953357bCa3f21d902e775f1F48",
+  },
+};
+
+// Get the address of the Pool Contract for the given environment
+export function poolAddress(environment: ethereum.Environment): string {
+  switch (environment) {
+    case ethereum.Environment.Local:
+      return addresses.pool.local;
+    case ethereum.Environment.Ropsten:
+      return addresses.pool.ropsten;
+  }
+}
 
-export function pool(signer: Signer): PoolContract {
-  return new PoolContract(signer, POOL_ADDRESS);
+export function pool(signer: Signer, address: string): PoolContract {
+  return new PoolContract(signer, address);
```

Comment:
yes good point!

Reply:
Actually, I prefer to keep it as is to avoid double dependencies on the site calls where these are used. I.e., if we drop this indirection, `wallet` needs to import `PoolContract` from `src/funding/contract` and `Erc20Factory` and `ERC20` from the Radicle Contracts.